### PR TITLE
[FT-WD-Berlin-W2D1-lab] Indrajeet Patil

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

File: src/movies.js [hunk 1]
```diff
@@ -1,25 +1,95 @@
...
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  return moviesArray.map((movie) => movie.director);
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  return moviesArray.filter((movie) => movie.director === "Steven Spielberg" && movie.genre.includes("Drama")).length;
```
Comment (matigobbi) [inline src/movies.js:15]: nice one line solution :D

File: src/movies.js [hunk 2]
```diff
@@ -1,25 +1,95 @@
...
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+  const averageScore = moviesArray.reduce((acc, movie) => {
+    return movie.score ? acc + movie.score : acc;
```
Comment (matigobbi) [inline src/movies.js:24]: 👏

**Review by matigobbi — COMMENTED** [review]
Very Good work Indra!

**Comment by IndrajeetPatil** [comment]
Thanks, @matigobbi! 😸

**Pull request closed without merging**